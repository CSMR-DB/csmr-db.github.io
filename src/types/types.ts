import React from 'react'

export type AnyExoticRefComponent<T> = React.ForwardRefExoticComponent<
  T & React.RefAttributes<any>
>

export type AnyExoticRefTargets =
  | ((instance: any) => void)
  | React.MutableRefObject<any>
  | null
