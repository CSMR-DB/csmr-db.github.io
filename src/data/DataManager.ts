import { ITool, toolsProvider } from './objects/toolsProvider'
import { themeProvider } from './objects/themeProvider'
import { DefaultTheme } from 'styled-components'
import { IRouteObject, routesProvider } from './objects/routesProvider'
import { ICourse, coursesProvider } from './objects/coursesProvider'
import { IContact, contactsProvider } from './objects/contactsProvider'
import {
  dynamicIconsProvider,
  IDynamicIcons,
} from './graphql/dynamicIconsProvider'
import {
  dynamicImagesProvider,
  IDynamicImages,
} from './graphql/dynamicImagesProvider'
import {
  experienceDataProvider,
  IExperienceData,
} from './graphql/experienceDataProvider'
import {
  gameWallpapersProvider,
  IGameWallpapers,
} from './graphql/gameWallpapersProvider'
import {
  ISeriesPosters,
  seriesPostersProvider,
} from './graphql/seriesPostersProvider'
import { ISiteMetadata } from '../types/interfaces'
import {
  ISkillsetData,
  skillsetDataProvider,
} from './graphql/skillsetDataProvider'
import { siteMetadataProvider } from './graphql/siteMetadataProvider'
import {
  IGraphicDesignData,
  graphicDesignDataProvider,
} from './graphql/graphicDesignDataProvider'
import {
  IPhotographySquares,
  photographySquaresProvider,
} from './graphql/photographySquaresProvider'
import {
  programmingDataProvider,
  IProgrammingData,
} from './graphql/programmingDataProvider'
import {
  photographyDataProvider,
  IPhotographyData,
} from './graphql/photographyDataProvider'

export abstract class AStaticDataManager {
  static contacts: IContact[]
  static courses: ICourse[]
  static routes: IRouteObject[]
  static theme: DefaultTheme
  static tools: ITool[]
  static siteMetadata: ISiteMetadata
  static skillsetData: ISkillsetData
  static experienceData: IExperienceData
  static programmingData: IProgrammingData
  static graphicDesignData: IGraphicDesignData
  static photographyData: IPhotographyData
  static gameWallpapers: IGameWallpapers
  static photographySquares: IPhotographySquares
  static seriesPosters: ISeriesPosters
  static dynamicImages: IDynamicImages
  static dynamicIcons: IDynamicIcons
}

/**
 * **Static Data Manager**
 *
 * Uses an abstract rather than an interface to implement, because TypeScript expects implemented interface to define the shape of an instance, not the class itself. Currently, using an abstract is the only way to get around that, as I can refer to it as
 * `typeof AStaticDataManager` anywhere I use it.
 *
 * Provides all 'static' data to Components through 1 single source of truth. No more duplicate-y GrahpQL queries with hidden differences between 2 files, but 1 predictable response.
 *
 * Everything else should remain consistent & predictable throughout the entire website to provide the best UX it can. I don't really need instances or a Singleton. The StaticDataManager shouldn't exist at runtime anyway.
 *
 * @Note:
 * StaticDataManager can only be used inside React Components, due to the use of the `useStaticQuery` hook in the Data properties.
 *
 * @FIXME:
 * I am not 100% satisfied with the current API "inconsistencies" when destructuring the data, but alas. That's a priority item for the next revision.
 */
export class StaticDataManager implements AStaticDataManager {
  static get routes(): IRouteObject[] {
    return routesProvider()
  }

  static get contacts(): IContact[] {
    return contactsProvider()
  }

  static get theme(): DefaultTheme {
    return themeProvider()
  }

  static get tools(): ITool[] {
    return toolsProvider()
  }

  static get courses(): ICourse[] {
    return coursesProvider()
  }

  static get siteMetadata(): ISiteMetadata {
    return siteMetadataProvider()
  }

  static get skillsetData(): ISkillsetData {
    return skillsetDataProvider()
  }

  static get experienceData(): IExperienceData {
    return experienceDataProvider()
  }

  static get programmingData(): IProgrammingData {
    return programmingDataProvider()
  }

  static get graphicDesignData(): IGraphicDesignData {
    return graphicDesignDataProvider()
  }

  static get photographyData(): IPhotographyData {
    return photographyDataProvider()
  }

  static get gameWallpapers(): IGameWallpapers {
    return gameWallpapersProvider()
  }

  static get photographySquares(): IPhotographySquares {
    return photographySquaresProvider()
  }

  static get seriesPosters(): ISeriesPosters {
    return seriesPostersProvider()
  }

  static get dynamicImages(): IDynamicImages {
    return dynamicImagesProvider()
  }

  static get dynamicIcons(): IDynamicIcons {
    return dynamicIconsProvider()
  }
}
