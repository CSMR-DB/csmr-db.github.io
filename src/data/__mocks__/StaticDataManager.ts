import { ExperienceData } from './../graphql/experienceDataProvider'
import { DynamicImages } from './../graphql/dynamicImagesProvider'
import { DynamicIcons } from './../graphql/dynamicIconsProvider'
import { ProjectsData } from '../graphql/projectsDataProvider'
import { SkillsetData } from '../graphql/skillsetDataProvider'
import { GameWallpapers } from '../graphql/gameWallpapersProvider'
import { SeriesPosters } from '../graphql/seriesPostersProvider'
import {
  SiteMetadata,
  SkillsetAllMarkdownRemark,
  ImageSharpAllFiles,
  ProjectsAllMarkdownRemark,
  ProjectMarkdownEdge,
  SkillsetMarkdownEdge,
  ExperienceAllMarkdownRemark,
} from '../../types/graphql.types'
import { PhotographySquares } from '../graphql/photographySquaresProvider'
import { PhotographyData } from '../graphql/photographyDataProvider'
import { Tool } from '../objects/toolsProvider'
import { AStaticDataManager } from '../StaticDataManager'
import { RouteObject } from '../objects/routesProvider'
import { Course } from '../objects/coursesProvider'
import { Contact } from '../objects/contactsProvider'
import { DefaultTheme } from 'styled-components'
import { mockSVGs } from './tools.mocks'

const mockImageCollection: ImageSharpAllFiles = {
  edges: [
    {
      node: {
        name: 'Photo',
        relativePath: '/photo.jpg',
        childImageSharp: {
          fluid: {
            aspectRatio: 1,
            sizes: '',
            src: '/photo.jpg',
            srcSet: '/photo.jpg',
          },
        },
      },
    },
  ],
}

const mockProgrammingSkillsetMdx: SkillsetAllMarkdownRemark = {
  edges: [
    {
      node: {
        frontmatter: {
          category: 'Programming',
          date: '1994-01-08T00:00:00+0000',
          title: 'React',
          excerpt: 'I love JSX',
          favorite: true,
          icon: '',
          skillColor: 'red',
          time: 1337,
        },
        generatedRoute: '/skillset/react',
      },
    },
    {
      node: {
        frontmatter: {
          category: 'Programming',
          date: '1994-01-08T00:00:00+0000',
          title: 'Jest',
          excerpt: 'It Jest needs to be done',
          favorite: true,
          icon: '',
          skillColor: 'blue',
          time: 1336,
        },
        generatedRoute: '/skillset/jest',
      },
    },
  ],
}

const mockProgrammingProjectsMdx: ProjectsAllMarkdownRemark = {
  edges: [
    {
      node: {
        frontmatter: {
          category: 'Programming',
          date: '1994-01-08T00:00:00+0000',
          title: 'Portfolio',
          excerpt:
            'Gatsby, TypeScript, React, GraphQL, GSAP, Styled-Components, Jest, Enzyme',
          favorite: true,
          tags: [
            'Gatsby',
            'TypeScript',
            'React',
            'GraphQL',
            'GSAP',
            'Styled-Components',
            'Jest',
            'Enzyme',
          ],
        },
        generatedRoute: '/projects/portfolio',
      },
    },
    {
      node: {
        frontmatter: {
          category: 'Programming',
          date: '1994-01-08T00:00:00+0000',
          title: 'Loopy',
          excerpt: 'I made a fantastic hula-loop once',
          favorite: true,
          tags: ['Hula', 'Loop', 'Perfection'],
        },
        generatedRoute: '/projects/portfolio',
      },
    },
  ],
}

const mockGraphicDesignProjectsMdx: ProjectsAllMarkdownRemark = {
  edges: [
    {
      node: {
        frontmatter: {
          category: 'Graphic Design',
          date: '1994-01-08T00:00:00+0000',
          title: 'Circle',
          excerpt: 'I drew a fantastic circle once',
          favorite: true,
          tags: ['Pencil', 'Shapes', 'Drawing'],
        },
        generatedRoute: '/projects/circle',
      },
    },
  ],
}

const mockGraphicDesignSkillsetMdx: SkillsetAllMarkdownRemark = {
  edges: [
    {
      node: {
        frontmatter: {
          category: 'Graphic Design',
          date: '1994-01-08T00:00:00+0000',
          title: 'Affinity Designer',
          excerpt: 'It has no proper geometry operations',
          favorite: false,
          icon: '',
          skillColor: 'green',
          time: 420,
        },
        generatedRoute: '/skillset/affinity_designer',
      },
    },
  ],
}

const mockExperienceMdx: ExperienceAllMarkdownRemark = {
  edges: [
    {
      node: {
        frontmatter: {
          title: 'React',
          backgroundColor: 'blue',
          type: 'Reactor',
          dateStart: '1994-01-08T00:00:00+0000',
          dateEnd: '1994-01-09T00:00:00+0000',
          excerpt: 'JSX is just fantastic',
        },
        generatedRoute: '',
      },
    },
    {
      node: {
        frontmatter: {
          title: 'Jest',
          backgroundColor: 'red',
          type: 'Jester',
          dateStart: '1994-01-08T00:00:00+0000',
          dateEnd: '1994-01-09T00:00:00+0000',
          excerpt: 'Jest a test',
        },
        generatedRoute: '',
      },
    },
  ],
}

export class StaticDataManager implements AStaticDataManager {
  static get theme(): DefaultTheme {
    return {
      palette: {
        dark: {
          normal: '#000',
          hover: '#222',
          active: '#444',
          lens: '#666',
        },
        light: {
          normal: '#fff',
          hover: '#ddd',
          active: '#bbb',
          lens: '#999',
        },
        first: { normal: '#f00', hover: '#d00', active: '#b00', lens: '#900' },
        second: {
          normal: '#0f0',
          hover: '#0d0',
          active: '#0b0',
          lens: '#090',
        },
        third: { normal: '#00f', hover: '#00d', active: '#00b', lens: '#009' },
        filter: {
          normal: 'saturation(4)',
          hover: 'saturation(4)',
          active: 'saturation(4)',
        },
      },
      breakpoints: {
        min: {
          smartphone: 'only screen and (min-width: 600px)',
          tablet: 'only screen and (min-width: 960px)',
          desktop: 'only screen and (min-width: 1200px)',
          ultrawide: 'only screen and (min-width: 1920px)',
        },
        max: {
          smartphone: 'only screen and (max-width: 600px)',
          tablet: 'only screen and (max-width: 960px)',
          desktop: 'only screen and (max-width: 1200px)',
          ultrawide: 'only screen and (max-width: 1920px)',
        },
      },
      fonts: {
        headings: 'Arial',
        copy: 'serif',
      },
    }
  }

  static get siteMetadata(): SiteMetadata {
    return {
      author: 'Jester',
      description: 'The TestRunner',
      image: '/',
      title: 'Jest',
      url: 'https://www.jest.testrunner',
    }
  }

  static get contacts(): Contact[] {
    return [
      {
        background: 'red',
        href: 'https://mock.test.nowhere',
        iconPath: '',
        text: 'Mock Option',
      },
    ]
  }

  static get courses(): Course[] {
    return [
      {
        category: 'Development',
        description: 'Experimenting with vanilla JS & jQuery',
        name: 'WebProgramming',
      },
    ]
  }

  static get routes(): RouteObject[] {
    return [
      {
        path: '/route',
        title: 'Route',
        boundKeys: ['R', 1],
      },
    ]
  }

  static get tools(): Tool[] {
    return [
      {
        name: 'React',
        svg: mockSVGs().RandomSampleSVG,
        color: 'red',
      },
    ]
  }

  static get photographyData(): PhotographyData {
    return {
      photographyData: mockImageCollection,
    }
  }

  static get photographySquares(): PhotographySquares {
    return {
      photographySquares: mockImageCollection,
    }
  }

  static get seriesPosters(): SeriesPosters {
    return {
      seriesPosters: mockImageCollection,
    }
  }

  static get gameWallpapers(): GameWallpapers {
    return {
      gameWallpapers: mockImageCollection,
    }
  }

  static get skillsetData(): SkillsetData {
    return {
      all: { ...mockProgrammingSkillsetMdx, ...mockGraphicDesignSkillsetMdx },
      top: {
        edges: [
          ...mockProgrammingSkillsetMdx.edges.filter(
            ({
              node: {
                frontmatter: { favorite },
              },
            }: SkillsetMarkdownEdge): boolean => favorite === true
          ),
          ...mockGraphicDesignSkillsetMdx.edges.filter(
            ({
              node: {
                frontmatter: { favorite },
              },
            }: SkillsetMarkdownEdge): boolean => favorite === true
          ),
        ],
      },
      allProgramming: mockProgrammingSkillsetMdx,
      allGraphicDesign: mockGraphicDesignSkillsetMdx,
    }
  }

  static get projectsData(): ProjectsData {
    return {
      all: { ...mockGraphicDesignProjectsMdx, ...mockProgrammingProjectsMdx },
      top: {
        edges: [
          ...mockGraphicDesignProjectsMdx.edges.filter(
            ({
              node: {
                frontmatter: { favorite },
              },
            }: ProjectMarkdownEdge): boolean => favorite === true
          ),
          ...mockProgrammingProjectsMdx.edges.filter(
            ({
              node: {
                frontmatter: { favorite },
              },
            }: ProjectMarkdownEdge): boolean => favorite === true
          ),
        ],
      },
      allProgramming: mockProgrammingProjectsMdx,
      allGraphicDesign: mockGraphicDesignProjectsMdx,
    }
  }

  static get experienceData(): ExperienceData {
    return {
      experienceData: mockExperienceMdx,
    }
  }

  static get dynamicIcons(): DynamicIcons {
    return {
      dynamicIcons: mockImageCollection,
    }
  }

  static get dynamicImages(): DynamicImages {
    return {
      dynamicImages: mockImageCollection,
    }
  }
}
