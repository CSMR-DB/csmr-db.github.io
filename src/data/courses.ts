import { theme } from './theme'

export interface ICourse {
  name: string
  description: string
  category:
    | 'Development'
    | 'Graphic Design'
    | 'Marketing'
    | 'Organizations'
    | 'Research'
    | 'Misc'
  color?: string
}

export const devCourses: ICourse[] = [
  {
    name: 'WebSkills',
    description: 'Just the absolute starter kit for working with HTML and CSS.',
    category: 'Development',
    color: theme.palette.dark.normal,
  },
  {
    name: 'Web Programming',
    description: `$('JavaScript'|'jQuery')`,
    category: 'Development',
    color: theme.palette.dark.normal,
  },
  {
    name: 'Realisation',
    description:
      'Using tools other than code to produce functional little apps. We worked with stuff like Appery.io and Layar.',
    color: theme.palette.dark.normal,
    category: 'Development',
  },
  {
    name: 'The Digital Workplace *',
    description:
      'A group project in about 8 groups of 4 to 5 students working together to create a prototype for a new Intranet for a client.',
    color: theme.palette.dark.normal,
    category: 'Development',
  },
  {
    name: 'Content Management Systems *',
    description: 'Building a website using Wordpress.',
    color: theme.palette.dark.normal,
    category: 'Development',
  },
]

export const marketingCourses: ICourse[] = [
  {
    name: 'Marketing & e-Business',
    description: 'An introduction course into digital marketing.',
    category: 'Marketing',
    // color: 'blue'
  },
  {
    name: 'Marketing & Communication',
    description: '',
    category: 'Marketing',
    // color: 'blue'
  },
  {
    name: 'Digital Methods',
    description:
      'Blogging about specific, current, topics regarding marketing.',
    category: 'Marketing',
    // color: 'blue'
  },
]

export const designCourses: ICourse[] = [
  {
    name: 'Workshop **',
    description:
      'We had 2 Workshops and these were not typical courses with an assessment, but instead were meant as an aid to other courses that relied on specific knowledge but were not designed to provide that in itself.',
    category: 'Graphic Design',
    // color: 'purple'
  },
  {
    name: 'Interaction Design',
    description:
      'Learning about core design principles like the 5-7 rule, visual aligment and more.',
    category: 'Graphic Design',
    // color: 'purple'
  },
  {
    name: 'Webdesign (with Axure)',
    description:
      'Apply what was taught at Interaction Design in a functional prototype built with Axure.',
    category: 'Graphic Design',
    // color: 'purple'
  },
  {
    name: 'Digital Storytelling',
    description:
      'Use your knowledge about Adobe Photoshop, Illustrator, After Effects and whatever else you want to produce an animated short.',
    category: 'Graphic Design',
    // color: 'purple'
  },
  {
    name: 'Film',
    description: 'Working with a camera, as opposed to just digital assets.',
    category: 'Graphic Design',
    // color: 'purple'
  },
]

export const organizationCourses: ICourse[] = [
  {
    name: 'Organisations & Communication',
    description: '',
    category: 'Organizations',
    // color: 'blue'
  },
  {
    name: 'Organisations & Change Management',
    description:
      'Say you need to implement a different workflow or new tools in an organization. How would you go about doing so?',
    category: 'Organizations',
    // color: 'blue'
  },
  {
    name: 'Processes & Information Supply',
    category: 'Organizations',
    description: '',
    // color: 'red'
  },
]

export const researchCourses: ICourse[] = [
  {
    name: 'Evaluation Research',
    description: 'Focus on qualitative research.',
    category: 'Research',
    // color: 'red'
  },
  {
    name: 'Testing & Measuring',
    description: 'Focus on quantitative research',
    category: 'Research',
    // color: 'red'
  },
]

export const miscCourses: ICourse[] = [
  {
    name: 'Doing New Things',
    description: '',
    category: 'Misc',
    // color: 'blue'
  },
  {
    name: 'Professional Skills',
    description: '',
    category: 'Misc',
    // color: 'blue'
  },
  {
    name: 'Legal Skills',
    description: '',
    category: 'Misc',
    // color: 'blue'
  },
  {
    name: 'Entrepreneurship',
    description: '',
    category: 'Misc',
    // color: 'blue'
  },
]

export const courses: ICourse[] = [
  ...devCourses,
  ...designCourses,
  ...marketingCourses,
  ...organizationCourses,
  ...researchCourses,
  ...miscCourses,
]
