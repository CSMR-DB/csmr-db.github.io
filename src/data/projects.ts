export interface IProject {
  image?: string
  video?: string
  title: string
  body: (string | { url: string; text: string })[]
  tags?: string[]
}
;[]

export const projects: IProject[] = [
  {
    title: 'Modern Family Fansite',
    image: 'projects/modernfamily.png',
    body: [
      `One of the first projects in year one that got me excited about
      webdevelopment. This project is just purely HTML5 + CSS3. Nothing too
      fancy, just the base necessities. The only thing that doesn't work is
      the actor pages. They don't exist due to deadlines at that time. Also,
      there are no links to the show's social pages. It was just a prototype
      to show off HTML & CSS. This project can also be found `,
      { url: 'http://modernfamily.csmr.nl', text: 'online' },
      '.',
    ],
    tags: ['HTML', 'CSS'],
  },
  {
    title: 'Cloudcademy',
    image: 'projects/cloudcademy_code.png',
    body: [
      `Cloudcademy was the big course assignment we were to work on to get
            a final grade. We were just told to make "a social media platform"
            for any subject we wanted. The subject itself didn't matter, as the
            goal was to learn JavaScript. In this project I chose to focus on
            jQuery, as it enabled me to get going faster. I made a login system
            (requirement), a post submission form with a dynamic display of
            images & videos and stored everything temporarily using the
            LocalStorage API to allow users to refresh a page and still preserve
            all the posts that had already been added. This was mainly done by
            concatenating strings of HTML, as shown in the image. For some
            reason the project doesn't run 100% properly anymore, so no live
            demo is available. The source code can still be found in `,
      {
        url: 'https://github.com/CSMR-DB/Cloudcademy',
        text: 'this GitHub repo',
      },
      '.',
    ],
    tags: ['HTML', 'CSS', 'JavaScript'],
  },
  {
    title: 'FOMO',
    video: 'https://www.youtube.com/embed/ZbANrSn5ZB0',
    body: [
      `In year one we got introduced to After Effects and I instantly fell in
          love. Soon enough, I found I needed more of a challenge. I wanted to do
          something in 3D. Through After Effects I found out about Cinema4D, and
          boy was it fun. I made the scenery in 3D and played around with lighting
          and camera tracks. The character in the video was animated in 2D, to
          give something of an "odd" feeling when watching it. I liked the style
          of side-scrolling games, and wanted to try to mimic that effect.`,
    ],
    tags: ['After Effects', 'Cinema4D', 'Illustrator'],
  },
  {
    title: 'Wordpress Theme',
    image: 'projects/wpthemeexample.png',
    body: [
      `At the end of year 1 there was this group project for a local
    students' guild centered around iceskating. I took the
    responsibility for the Wordpress website design upon me, learning a
    bit about the PHP used by Wordpress to develop a theme that fully
    utilized the Customizer API. Unfortunately, we weren't chosen to
    publish our design concept.`,
    ],
    tags: ['HTML', 'CSS', 'JavaScript', 'PHP'],
  },
  {
    title: 'Abstract Dataviz',
    image: 'projects/dataroute.png',
    body: [
      `During my year 4 minor we had one major assignment: Take any form of
    data you can find on your route from home to college. I chose mobile
    network speed, took 20 points of measure, fiddled up some concepts
    and eventually landed on this very abstract piece. Inspired by the
    typically Dutch `,
      {
        url: 'https://nl.wikipedia.org/wiki/Paddenstoel_(wegwijzer)',
        text: '"route paddenstoel"',
      },
      ` and an ordinary colorful mushroom, flattened as if you were viewing
    it top-down with the data displayed on it in the LTR writing system.`,
    ],
    tags: ['Illustrator'],
  },
  {
    title: 'Video Project VUMC',
    video: 'https://www.youtube.com/embed/OYVxTuwnx-c',
    body: [
      `At the end of the year 4 minor there was this big assignment: make a
    datavisualization for a client according to their needs &
    requirements. The entire group of students was split up in smaller
    teams of 3, each working for a different client with different
    requirements. Our objective: make an animated short, telling the
    story about our research and findings. One that we could use in
    schools and one that is interesting for both students and teachers.
    Make sure to get our message "we need to come together to find a
    solution" across. We created concepts, sketches and discussed ideas
    about doing a voice-over. We made a version using a voice-over, but
    were asked to make a version with subtitles instead. I took it upon
    myself to create the entire animation in After Effects and find a
    suiting (& free) piece of background music. The end result: a
    composition with consistency at the center, and smooth transitions
    to highlight important sections.`,
    ],
    tags: ['After Effects', 'Illustrator'],
  },
  {
    title: 'PHYX API',
    image: 'projects/phyx_api.png',
    body: [
      `This is my first major project stepping out of front-end
    development. Everything I have made up until this point has
    focussed, in some way, on visual design. The only visual aspect of
    this project is the GraphiQL portal that's made available to
    communicate with the server. The stack consists of Apollo Server
    (GraphQL), TypeScript and a Dependency Injection Container. This is
    a project for me to experiment with architecture, TDD & SOLID
    principles. I still have a lot to learn. Full code for this project
    is `,
      {
        url: 'https://github.com/CSMR-DB/phyx-api',
        text: 'available on GitHub',
      },
      `.`,
    ],
    tags: ['TypeScript', 'GraphQL'],
  },
]
