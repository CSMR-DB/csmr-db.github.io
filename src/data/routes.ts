export interface IRouteObject {
  path: string
  title: string
}

export const routes: IRouteObject[] = [
  {
    path: '/about',
    title: 'About',
  },
  {
    path: '/skillset',
    title: 'Skillset',
  },
  {
    path: '/projects',
    title: 'Projects',
  },
  {
    path: '/contact',
    title: 'Contact',
  },
]
