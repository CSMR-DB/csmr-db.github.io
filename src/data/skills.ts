import { StringOrUrlArray } from '../types/StringOrUrlObject'

export interface ISkill {
  title: string
  level: number
  description: StringOrUrlArray
  time: number
}

export interface ISkills {
  programming: ISkill[]
  graphics: ISkill[]
}

export const skills: ISkills = {
  programming: [
    {
      title: 'HTML',
      level: 4,
      description:
        "HTML, along with CSS, is obviously used in any front-end JavaScript framework. Either plainly, or through JSX. I find HTML's default margins on elements stupidly annoying and always require some sort of a 'reset' CSS file. Also: article-header-main-aside-footer. I'm just glad I wasn't around during the whole table-layout-everything-fad.",
      time: 3000,
    },
    {
      title: 'CSS',
      level: 4,
      description:
        "I love CSS and everything you can do with it nowadays without having to resort to JavaScript. Most notably, the CSS Grid, Flex and Filter specs. But also clip-path seems pretty cool. I always use some sort of preprocessor, be it Stylus or something like styled-components or Emotion for React (or whatever is available CSS-in-JS-wise. I don't like having to worry about naming things following any naming scheme like BEM). Not a big fan of using UI frameworks like Bootstrap or Material UI, as I feel it limits creativity.",
      time: 3000,
    },
    {
      title: 'JavaScript',
      level: 4,
      description:
        'I feel adequate using everything above ES6, but not about working with polyfills / older versions of JavaScript. I can work with the class syntax, or just plain Objects and Functions. I get Promises and Async / Await, but I feel like I could still learn how to apply it in better ways. I have played with Web Components and I like the idea. There just needs to be 1 definitive spec & implementation across (modern) browsers.',
      time: 2000,
    },
    {
      title: 'TypeScript',
      level: 3,
      description: [
        "I absolutely love it. I understand TypeScript's limits but having the compile-time safety has proven to be a huge time-saver, especially when using the strictest tsconfig imagineable. Combining the power of ",
        {
          url:
            'https://github.com/CSMR-DB/phyx-api/blob/master/src/tiny-di/tiny-di.ts',
          text: 'Reflect-metadata with Decorators to create a tiny DI system',
        },
        ' is awesome. (Thanks to ',
        {
          url:
            'https://medium.com/@OlegVaraksin/minimalistic-dependency-injection-di-container-in-typescript-2ce93d1c303b',
          text: 'this Medium post',
        },
        ').',
      ],
      time: 600,
    },
    {
      title: 'GraphQL',
      level: 2,
      description:
        'Yes! Please, use it more. I think the developer experience of using GraphQL is absolutely amazing. Having control over WHAT you want to fetch, preventing over- / under-fetching of data is just marvellous. And you can wrap REST APIs in GraphQL, forcing the server to do the heavy lifting of dealing with over- & under-fetching of data and returning only the essential data to the client. Getting an object with 200 lines of data when I only want about 20 is NOT fun.',
      time: 400,
    },
    {
      title: 'React',
      level: 3,
      description:
        'As with all view-libraries I believe that one should be able to replace the layer at any point in time. Business logic has no place in Components and should be extracted to services, so that when you might swap out the view layer, you could just rely on those exact services to provide you with the data you want to display.',
      time: 1000,
    },
    {
      title: 'Angular',
      level: 2,
      description:
        "I love the complexity, but haven't used it much at all. It's built with TypeScript, so that's a big plus. It encourages the use of Services to hold business logic more so than React does. Also, Angular taught me the idea of Dependency Injection.",
      time: 200,
    },
    {
      title: 'Vue',
      level: 1,
      description: [
        "Hardly ever really touched Vue, but I'm really interested to learn more about it. The only thing I ever made was ",
        {
          url: 'https://codepen.io/CSMR/pen/wjRaqe',
          text: 'a single-page prototype',
        },
        ' for an assignment in year 3.',
      ],
      time: 100,
    },
    {
      title: 'MongoDB',
      level: 2,
      description:
        "MongoDB being provided by default in MeteorJS' stack has made it the first DB engine I have worked with. Reading the JSON-like format used in Collections has made it easy to work with. I've used Mongoose as the NodeJS driver to create strongly typed schemas and to .populate() documents in a somewhat relational manner, which worked out well for that use-case.",
      time: 400,
    },
    {
      title: 'Jest',
      level: 2,
      description:
        "Let's just say that thus far I've only really used Jest as my JavaScript testing framework, extended by ts-jest. Other frameworks might work just as well, I would imagine. Thus far, I have mainly used it to test business logic. Not much in the front-end world.",
      time: 400,
    },
    {
      title: 'Git',
      level: 1,
      description:
        "I know about and understand the purpose of version control and find it infuriating to see big (game) publishers not using it properly. But personally, I have only really used Git for personal projects. I haven't dealt with branches or any form of consistency in commits, as I've never really felt the need to. I haven't yet collaborated with anyone or had someone to code review with, so my commits tend to be big and messy. This is just for personal projects where I'm all about experimenting and refactoring constantly.",
      time: 20,
    },
    {
      title: 'D3',
      level: 2,
      description:
        'During my internship I got to know the basics of D3.js and played around with componentizing it for use with React. I have only used the SVG variant thus far, and would love to experiment with Canvas.',
      time: 400,
    },
  ],
  graphics: [
    {
      title: 'Adobe Illustrator',
      level: 3,
      description:
        "You need me to help out with design? No problem at all. I consider myself to be able to at least take on someone else's duties during their time off. Image Trace is my secret sauce.",
      time: 400,
    },
    {
      title: 'Adobe After Effects',
      level: 2,
      description:
        'I have only really used it to make short animations, or small compilations of videos for courses. I know that the the quality of source footage is crucial when it comes to making a decent composition. Highly enjoy working with it, for sure.',
      time: 200,
    },
    {
      title: 'Adobe InDesign',
      level: 2,
      description:
        "It was pretty much a requirement in college to create all kinds of documentation using InDesign. The possibilities it gives you with Master Pages and the flexibility with TOC generation is absolutely amazing. The only thing aside from more conventional documents I've ever really made is a basic Styleguide for a fictional company as part of a course assignment.",
      time: 100,
    },
  ],
}
