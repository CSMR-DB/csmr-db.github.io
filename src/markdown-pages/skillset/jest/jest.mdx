---
category: 'Programming'
level: 2
skillColor: '#C63D14'
time: 400
title: 'Jest'
excerpt: "I mean, it's one of many testing frameworks. I mainly just like the logo."
icon: <JestSVG/>
---
