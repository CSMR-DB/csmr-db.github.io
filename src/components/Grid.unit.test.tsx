import React from 'react'
import { ThemeProvider } from 'styled-components'
import { create, ReactTestRenderer } from 'react-test-renderer'

import { RTRTree } from '../types/jest.types'
import { StaticDataManager } from '../data/StaticDataManager'

import { Grid } from './Grid'

describe('Layout', (): void => {
  it('should have the following style rules by default', (): void => {
    const comp: ReactTestRenderer = create(
      <ThemeProvider theme={StaticDataManager.theme}>
        <Grid>
          <div>Test</div>
          <div>These</div>
          <div>Divs</div>
        </Grid>
      </ThemeProvider>
    )
    const json: RTRTree = comp.toJSON()

    expect(comp).toMatchSnapshot()

    expect(json).toHaveStyleRule('display', 'grid')
    expect(json).toHaveStyleRule(
      'grid-template-columns',
      'repeat(auto-fill,minmax(16rem,1fr))'
    )
    expect(json).toHaveStyleRule('gap', 'calc(1rem / 2)')
    expect(json).toHaveStyleRule('align-items', 'initial')
    expect(json).toHaveStyleRule('grid-template-rows', 'auto')

    /** Media query rules */
    expect(json).toHaveStyleRule('gap', '1rem', {
      media: StaticDataManager.theme.breakpoints.min.smartphone,
    })
    expect(json).toHaveStyleRule('grid-template-columns', 'auto', {
      media: StaticDataManager.theme.breakpoints.min.smartphone,
    })
  })

  it('should have the following style rules on screen sizes tablet and up', (): void => {
    const comp: ReactTestRenderer = create(
      <ThemeProvider theme={StaticDataManager.theme}>
        <Grid $rows={3} $columns={3} $gap={'2rem'}>
          <div>Test</div>
          <div>These</div>
          <div>Divs</div>
        </Grid>
      </ThemeProvider>
    )
    const json: RTRTree = comp.toJSON()

    expect(comp).toMatchSnapshot()

    expect(json).toHaveStyleRule('grid-template-rows', 'repeat(3,1fr)')
    expect(json).toHaveStyleRule('gap', 'calc(2rem / 2)')

    /** Media query rules */
    expect(json).toHaveStyleRule('grid-template-columns', 'repeat(3,1fr)', {
      media: StaticDataManager.theme.breakpoints.min.smartphone,
    })
    expect(json).toHaveStyleRule('gap', '2rem', {
      media: StaticDataManager.theme.breakpoints.min.smartphone,
    })
  })
})
