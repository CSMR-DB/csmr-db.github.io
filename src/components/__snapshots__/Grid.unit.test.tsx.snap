// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should have the following style rules by default 1`] = `
.c0 {
  display: grid;
  margin: 0 auto;
  height: auto;
  max-width: auto;
  gap: calc(1rem / 2);
  grid-template-columns: repeat(auto-fill,minmax(16rem,1fr));
  grid-template-rows: auto;
  -webkit-align-items: initial;
  -webkit-box-align: initial;
  -ms-flex-align: initial;
  align-items: initial;
}

@media only screen and (min-width:600px) {
  .c0 {
    grid-template-columns: auto;
    gap: 1rem;
  }
}

<div
  className="c0"
>
  <div>
    Test
  </div>
  <div>
    These
  </div>
  <div>
    Divs
  </div>
</div>
`;

exports[`Layout should have the following style rules on screen sizes tablet and up 1`] = `
.c0 {
  display: grid;
  margin: 0 auto;
  height: auto;
  max-width: auto;
  gap: calc(2rem / 2);
  grid-template-columns: repeat(auto-fill,minmax(16rem,1fr));
  grid-template-rows: repeat(3,1fr);
  -webkit-align-items: initial;
  -webkit-box-align: initial;
  -ms-flex-align: initial;
  align-items: initial;
}

@media only screen and (min-width:600px) {
  .c0 {
    grid-template-columns: repeat(3,1fr);
    gap: 2rem;
  }
}

<div
  className="c0"
>
  <div>
    Test
  </div>
  <div>
    These
  </div>
  <div>
    Divs
  </div>
</div>
`;
